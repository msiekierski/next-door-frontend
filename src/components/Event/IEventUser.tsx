interface IEventUser {
    idAccount: number,
    name: string,
    surname: string
}

export default IEventUser;