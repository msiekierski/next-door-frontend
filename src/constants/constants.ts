export const ANNOUNCEMENT_TYPE_COMMUNAL: number = 1;
export const ANNOUNCEMENT_TYPE_ADMINISTRATIVE: number = 2;
export const ANNOUNCEMENT_TYPE = "announcement";
export const EVENT_TYPE = "event";
export const ADVERT_TYPE_UNPAID: number = 1;
export const ADVERT_TYPE_PAID: number = 2;
export const ADVERT_BOTTOM_PAYMENT: number = 100;
export const ADVERT_PER_ONE_PAYMENT: number = 10;
