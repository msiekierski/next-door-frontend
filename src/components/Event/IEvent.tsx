import IEventUser from "./IEventUser";

interface IEvent {
  idEvent: number;
  type: "event";
  idCreator: number;
  title: string;
  description: string;
  creationDate: string;
  eventDate: string;
  idAssoc: number;
  creatorName?: string;
  creatorSurname?: string;
  users: Array<IEventUser>;
}

export default IEvent;
