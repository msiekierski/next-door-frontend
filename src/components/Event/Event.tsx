import React, { FunctionComponent, MouseEvent, useContext, useState, useRef } from "react";
import { Card, Button, Form } from "react-bootstrap";
import { AiOutlineEdit } from "react-icons/ai";
import { IUser } from "../Login/IUser";
import { deleteAnnouncement, putAnnouncement } from "../../API/announcements";
import IEvent from "./IEvent";
import { UserContext } from "../Login/UserContext";
import { deleteEvent, putEvent } from "../../API/events";
import oracleDateToInputDate from "../../utils/DateConverter";

export type Props = IEvent & {
  removeEvent: Function;
  updateEvent: Function;
};

const Event: FunctionComponent<Props> = ({
  idEvent,
  idCreator,
  title,
  description,
  creationDate,
  eventDate,
  removeEvent,
  updateEvent
}) => {
  const [isEditing, setIsEditing] = useState(false);
  const [titleEdit, setTitleEdit] = useState(title);
  const [descriptionEdit, setDescriptionEdit] = useState(description);
  const user = useContext(UserContext);
  const inputEventDate = useRef<HTMLInputElement>(null);

  const handleJoinClick = (e: MouseEvent) => {
    e.preventDefault();
  };

  const handleShowParticipiants = (e: MouseEvent) => {
    e.preventDefault();
  };

  const handleCancelEditClick = (e: MouseEvent) => {
    e.preventDefault();
    clearEdit();
    setIsEditing(false);
  };

  const handleSaveEditClick = async (e: MouseEvent) => {
    e.preventDefault();
    await putEvent(
      idEvent,
      titleEdit,
      descriptionEdit,
      inputEventDate?.current?.value ? inputEventDate.current.value : ""
    );
    updateEvent(
      idEvent,
      titleEdit,
      descriptionEdit,
      inputEventDate?.current?.value ? inputEventDate.current.value : ""
    );
    clearEdit();
    setIsEditing(false);
  };

  const clearEdit = () => {
    setDescriptionEdit(description);
    setTitleEdit(title);
  };

  const handleDeleteEditClick = async (e: MouseEvent) => {
    e.preventDefault();
    await deleteEvent(idEvent);
    setIsEditing(false);
    removeEvent(idEvent);
  };
  return (
    <Card className={`mt-3`}>
      <Card.Header className={`bg-info`} />
      <Card.Body>
        <Card.Title className={`d-flex justify-content-between`}>
          <span>
            {!isEditing ? (
              title
            ) : (
              <Form.Control type="text" value={titleEdit} onChange={(e) => setTitleEdit(e.target.value)} />
            )}
          </span>
          <div className="flex-column">
            <Card.Subtitle className={`text-right mb-3`}>
              {user?.idAccount === idCreator && !isEditing && (
                <span className={`btn`}>
                  <AiOutlineEdit onClick={() => setIsEditing(!isEditing)} />{" "}
                </span>
              )}
              {user?.idAccount === idCreator && isEditing && (
                <div className={`d-flex justify-content-end`}>
                  <Button variant="danger" className="mr-2" onClick={handleDeleteEditClick}>
                    Delete
                  </Button>
                  <Button variant="warning" className="mr-2" onClick={handleCancelEditClick}>
                    Cancel
                  </Button>
                  <Button variant="primary" onClick={handleSaveEditClick}>
                    Save
                  </Button>
                </div>
              )}
            </Card.Subtitle>
            {!isEditing && (
              <>
                <Card.Subtitle className={`text-muted text-right mb-2`}>Takes place on:</Card.Subtitle>
                <Card.Subtitle className={`text-muted text-right`}>
                  <u>{new Date(eventDate).toLocaleDateString()}</u>
                </Card.Subtitle>
              </>
            )}
          </div>
        </Card.Title>
        {isEditing && (
          <Card.Text>
            <input defaultValue={eventDate} ref={inputEventDate} className="form-control w-25" required type="date" />
          </Card.Text>
        )}

        <Card.Text>
          {!isEditing ? (
            description
          ) : (
            <Form.Control type="text" value={descriptionEdit} onChange={(e) => setDescriptionEdit(e.target.value)} />
          )}
        </Card.Text>
        <Card.Subtitle className={`text-muted text-right`}>{new Date(creationDate).toLocaleDateString()}</Card.Subtitle>
      </Card.Body>
      <Card.Footer className={`d-flex justify-content-between`}>
        <Card.Link href={``} onClick={handleJoinClick}>
          Join
        </Card.Link>
        <Card.Link href={``} onClick={handleShowParticipiants}>
          0 participiants
        </Card.Link>
      </Card.Footer>
    </Card>
  );
};
export default Event;
