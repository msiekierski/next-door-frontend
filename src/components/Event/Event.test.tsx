import React from "react";
import renderer from "react-test-renderer";
import { EVENT_TYPE } from "../../constants/constants";
import Event from "./Event";
import IEvent from "./IEvent";

type Props = IEvent & {
  removeFeed: Function;
  updateFeed: Function;
};

const event: Props = {
  idEvent: 0,
  idCreator: 1,
  title: "New event",
  description: "Come, join us!",
  creationDate: new Date().toString(),
  eventDate: new Date().setDate(new Date().getDate() + 10).toString(),
  idAssoc: 1,
  type: EVENT_TYPE,
  removeFeed: () => {},
  updateFeed: () => {},
};

it("renders correctly", () => {
  const tree = renderer.create(<Event {...event} />);
  expect(tree).toMatchSnapshot();
});
