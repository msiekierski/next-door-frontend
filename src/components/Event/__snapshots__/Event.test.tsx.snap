// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-3 card"
>
  <div
    className="bg-info card-header"
  />
  <div
    className="card-body"
  >
    <div
      className="d-flex justify-content-between card-title h5"
    >
      <span>
        New event
      </span>
      <div
        className="flex-column"
      >
        <div
          className="text-right mb-3 card-subtitle h6"
        />
        <div
          className="text-muted text-right mb-2 card-subtitle h6"
        >
          Takes place on:
        </div>
        <div
          className="text-muted text-right card-subtitle h6"
        >
          <u>
            Invalid Date
          </u>
        </div>
      </div>
    </div>
    <p
      className="card-text"
    >
      Come, join us!
    </p>
    <div
      className="text-muted text-right card-subtitle h6"
    >
      2021-1-22
      <br />
      1:31:29
    </div>
  </div>
  <div
    className="d-flex justify-content-between card-footer"
  >
    <a
      className="card-link"
      href=""
      onClick={[Function]}
    >
      Join
    </a>
    <a
      className="card-link"
      href=""
      onClick={[Function]}
    >
      0 participiants
    </a>
  </div>
</div>
`;
